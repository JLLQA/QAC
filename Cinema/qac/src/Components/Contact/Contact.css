#potato{
    font-weight: 500;
}

.formGroup{
    padding-top: 10px;
    padding-bottom: 10px;
}

.alertOne{
    line-height: auto;
    font-size: 15px;
    color: palevioletred;
 
}

input[type=text] {
    padding: 10px;
    width: 90%;
    max-width: 700px;
    margin: auto;
    border: 2px solid rgb(59, 124, 124);
  }

#subject{
    padding: 10px;
    width: 90%;
    max-width: 700px;
    margin: auto;
    border: 2px solid rgb(59, 124, 124);
    text-align: center;
}

#label{
    text-align: right;
}

#messageBox{
    padding: 10px;
    width: 90%;
    max-width: 800px;
    height: 150px;
    margin: auto;
    border: 2px solid rgb(59, 124, 124);
    text-align: center;
}

#submitButton{
    padding: 10px;
    width: 90%;
    max-width: 800px;
    background-color: slategray;
    border: 2px solid rgb(255, 255, 255);
}

#submitButton:hover{
    background-color: orange;
}





