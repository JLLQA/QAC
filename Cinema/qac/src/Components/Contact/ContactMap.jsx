import GoogleMapReact from "google-map-react";
import { Button } from "reactstrap";

const ContactMap = () => {
    const AnyReactComponent = ({ text }) => <div>{text}</div>;

    const defaultProps = {
        center: {
            lat: 59.95,
            lng: 30.33
        },
        zoom: 11
    };

    return (

        <div className="container" align="center">
            <h2 >Where We Are</h2>
            <div className="row justify-content-md-center">

                <div className="col-md-auto" >
                    <h4>Our Address</h4>
                    <p>8, St James's Bldg</p>
                    <p>61-95 Oxford St</p>
                    <p>Manchester</p>
                    <p>M1 6FQ</p>
                    <p>United Kingdom</p>
                    <Button type="button" onClick={(e) => {e.preventDefault();
                            window.location.href = '';
                        }}>Click for more Location Info</Button>
                </div>


                <div className="col col-lg-6" style={{ height: '40vh', width: '50%' }}>
                    <GoogleMapReact
                        bootstrapURLKeys={{ key: "AIzaSyDgxqUYRNs1Zyd7FXeovnBzHrFmzB-7Q0I" }}
                        defaultCenter={defaultProps.center}
                        defaultZoom={defaultProps.zoom}
                    >
                        <AnyReactComponent //find long lat of place QAC cinema
                            lat={59.955413}
                            lng={30.337844}
                            text="My Marker"
                        />
                    </GoogleMapReact>
                </div>
            </div>
        </div>
    )

}

export default ContactMap;