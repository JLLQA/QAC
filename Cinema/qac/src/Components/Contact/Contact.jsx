import emailjs from 'emailjs-com';
import { useForm } from "react-hook-form";
import './Contact.css';
import ContactMap from './ContactMap';
import Navbar from '../Navbar/Navbar';
import { Container } from 'reactstrap';

const ContactUs = () => {

    const { register, handleSubmit, formState: { errors } } = useForm();

    const onSubmit = (data, e) => {
        console.log(data, e);
        emailjs.sendForm('service_z6tey7i', 'template_ly9kskf', e.target, 'user_F3eP1aV9v0QtqRuME02vV')
            .then((data) => {
                console.log('SUCCESS!', data.status, data.text);
                e.target.reset();
                alert("Your form has been submitted successfully");
            })
            .catch((errors) => {
                console.log('Failed...', errors.text);
            })
    };

    return (
        <>
            <div>
                <Navbar />
            </div>
            <div>
                <Container>
                    <h1 align="center">Contact</h1>
                    <ContactMap />
                    <br />
                    <div id="potato" align="center" >
                        <h2>Send us an Email</h2>
                        <h4>Please fill out the form below and we will reply by email</h4>
                        <form className="contact-form" onSubmit={handleSubmit(onSubmit)}>
                            <div>
                                <div className="formGroup">
                                    <div>
                                        <label className="label">Name:</label>
                                    </div>
                                    <div >
                                        <input type="text" {...register("Name", { required: true, minLength: 3, maxLength: 30 })} />
                                        <div className="alertOne">
                                            {errors.Name?.type === 'required' && <p>  A name is required  </p>}
                                            {errors.Name?.type === 'minLength' && <p>  3 Characters required  </p>}
                                            {errors.Name?.type === 'maxLength' && <p>  30 Characters exceeded  </p>}
                                        </div>
                                    </div>
                                </div>
                                <div className="formGroup">
                                    <div>
                                        <label>Email:</label>
                                    </div>
                                    <div >
                                        <input type="text" {...register("Email", { required: true, pattern: /\S+@\S+\.\S+/ })} />
                                        <div className="alertOne">
                                            {errors.Email?.type === 'required' && <p>An email is required</p>}
                                            {errors.Email?.type === 'pattern' && <p>Enter a valid email</p>}
                                        </div>
                                    </div>
                                </div>
                                <div className="formGroup">
                                    <div>
                                        <label for="subject" >Subject:</label>
                                    </div>
                                    <div>
                                        <select id="subject" {...register("Subject", { required: "select one option" })}>
                                            <option disabled selected value="">-- Select a Subject --</option>
                                            <option value="Movies">Movies</option>
                                            <option value="Tickets">Ticket Enquiries</option>
                                            <option value="Location">Location</option>
                                            <option value="Other">Other</option>
                                        </select>
                                        <div className="alertOne">
                                            {errors.Subject && <p>Select a Subject</p>}
                                        </div>
                                    </div>
                                </div>
                                <div className="formGroup">
                                    <div>
                                        <label>Message:</label>
                                    </div>
                                    <div >
                                        <textarea id="messageBox" {...register("Message", { required: true, minLength: 10, maxLength: 1000 })} />
                                        <div className="alertOne">
                                            {errors.Message?.type === 'required' && <p>A message is required</p>}
                                            {errors.Message?.type === 'minLength' && <p>10 Characters required</p>}
                                            {errors.Message?.type === 'maxLength' && <p>1000 Characters exceeded</p>}
                                        </div>
                                    </div>
                                </div>
                                <div className="formGroup">
                                    <input id="submitButton" type="submit" value=" Submit Form " />
                                </div>
                                <div>
                                    <button type="reset" id="resetButton" value="reset"> Reset Form </button>
                                </div>
                            </div>
                        </form>
                    </div>
                </Container>
            </div>
        </>
    );
}

export default ContactUs;