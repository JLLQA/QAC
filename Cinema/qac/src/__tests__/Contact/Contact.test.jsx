import '@testing-library/jest-dom';
import { fireEvent,getByText,render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Contact from '../../Components/Contact/Contact';

describe('test for contact page', () => {

    test('CONTACT heading text is present', () => {
        render(<Contact />);
        const linkElement = screen.getByText("CONTACT");
        expect(linkElement).toBeInTheDocument();
    });

    test("testing hrefs to contain links", () => {
        render(<Contact />);
        expect(screen.getByRole('link', { name: /Click for more Travel Info/i }).closest('a')).toHaveAttribute('href', 'http://localhost:3000/gettingthere');
        expect(screen.getByRole('button', { name: /Reset Form/i })).toHaveAttribute('type', 'reset');
    });

    test("It matches the snapshot", () => {
        const tree = renderer.create(<Contact />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("testing the form", () => {

        const handleSubmit = jest.fn();
      const formTest = render(
            <Contact handleSubmit={handleSubmit} />
        );
        
        const nameInput = screen.getByPlaceholderText('Enter name here');
        const emailInput = screen.getByPlaceholderText('Enter email here');
              
        const messageInput = screen.getByPlaceholderText('Enter message here');
        
        fireEvent.change(nameInput, { target: { value: "abc" } });
        fireEvent.change(emailInput, { target: { value: "abc@qa.com" } });
        
        fireEvent.click(screen.getByText("-- Select a Subject --"));
        fireEvent.click(screen.getByText("Movies"));
        
        fireEvent.change(messageInput, { target: { value: "hello" } });

        expect(nameInput.value).toBe("abc");
        expect(emailInput.value).toBe("abc@qa.com");
        expect(messageInput.value).toBe("hello");
           
    });
});