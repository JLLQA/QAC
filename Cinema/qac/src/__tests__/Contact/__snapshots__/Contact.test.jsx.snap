// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for contact page It matches the snapshot 1`] = `
<div>
  <div
    className="container-fluid"
    id="dropped-box"
  >
    <div
      className="container"
    >
      <h1>
        CONTACT
      </h1>
      <div
        align="center"
        className="container"
      >
        <br />
        <h2>
          Where We Are
        </h2>
        <br />
        <div
          className="row justify-content-md-center"
        >
          <div
            className="col-md-auto"
          >
            <h3>
              Our Address
            </h3>
            <p>
              8th Floor St James Buildings
              <br />
              79 Oxford Street
              <br />
              Manchester
              <br />
              M1 6FQ
            </p>
          </div>
          <div
            className="col col-lg-6"
            style={
              Object {
                "height": "40vh",
                "width": "50%",
              }
            }
          >
            <div
              onClick={[Function]}
              onMouseDownCapture={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "height": "100%",
                  "margin": 0,
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "margin": 0,
                    "padding": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "height": "50%",
                    "left": "50%",
                    "margin": 0,
                    "padding": 0,
                    "position": "absolute",
                    "top": "50%",
                    "width": "50%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "height": 0,
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": 0,
                      }
                    }
                  >
                    <div>
                      QA Cinema
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        align="center"
      >
        <a
          aria-label={null}
          className="btn btn-secondary"
          href="http://localhost:3000/gettingthere"
          id="infoButton"
          onClick={[Function]}
          type="button"
        >
          Click for more Travel Info
        </a>
        <br />
        <br />
      </div>
      <div
        align="center"
      >
        <h2>
          Send us an Email
        </h2>
        <h4>
          Please fill out the form below and we will reply by email
        </h4>
        <form
          className="contact-form"
          onSubmit={[Function]}
        >
          <div>
            <div
              className="formGroup"
            >
              <div>
                <label>
                  Name:
                </label>
              </div>
              <div>
                <input
                  id="contact"
                  name="Name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter name here"
                  type="text"
                />
                <div />
              </div>
            </div>
            <div
              className="formGroup"
            >
              <div>
                <label>
                  Email:
                </label>
              </div>
              <div>
                <input
                  id="contact"
                  name="Email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter email here"
                  type="text"
                />
                <div />
              </div>
            </div>
            <div
              className="formGroup"
            >
              <div>
                <label>
                  Subject:
                </label>
              </div>
              <div>
                <select
                  id="subject"
                  name="Subject"
                  onBlur={[Function]}
                  onChange={[Function]}
                >
                  <option
                    disabled={true}
                    selected={true}
                    value=""
                  >
                    -- Select a Subject --
                  </option>
                  <option
                    value="Movies"
                  >
                    Movies
                  </option>
                  <option
                    value="Tickets"
                  >
                    Ticket Enquiries
                  </option>
                  <option
                    value="Location"
                  >
                    Location
                  </option>
                  <option
                    value="Other"
                  >
                    Other
                  </option>
                </select>
                <div />
              </div>
            </div>
            <div
              className="formGroup"
            >
              <div>
                <label>
                  Message:
                </label>
              </div>
              <div>
                <textarea
                  id="messageBox"
                  name="Message"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter message here"
                />
                <div />
              </div>
            </div>
            <div
              className="formGroup"
            >
              <input
                id="submitButton"
                type="submit"
                value=" Submit Form "
              />
            </div>
            <div>
              <button
                id="resetButton"
                type="reset"
                value="reset"
              >
                 Reset Form 
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
